File: src/test/java/cz/vhromada/catalog/web/mapper/ShowMapperTest.java
package cz.vhromada.catalog.web.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.entity.Show;
import cz.vhromada.catalog.web.common.ShowUtils;
import cz.vhromada.catalog.web.fo.ShowFO;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link Show} and {@link ShowFO}.
 *
 * @author Vladimir Hromada
 */
class ShowMapperTest {

    /**
     * Mapper for shows
     */
    private ShowMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(ShowMapper.class);
    }

    /**
     * Test method for {@link ShowMapper#map(Show)}.
     */
    @Test
    void map() {
        final Show show = ShowUtils.getShow();

        final ShowFO showFO = mapper.map(show);

        ShowUtils.assertShowDeepEquals(show, showFO);
    }

    /**
     * Test method for {@link ShowMapper#map(Show)} with null show.
     */
    @Test
    void map_NullShow() {
        assertThat(mapper.map(null)).isNull();
    }

    /**
     * Test method for {@link ShowMapper#mapBack(ShowFO)}.
     */
    @Test
    void mapBack() {
        final ShowFO showFO = ShowUtils.getShowFO();

        final Show show = mapper.mapBack(showFO);

        ShowUtils.assertShowDeepEquals(showFO, show);
    }

    /**
     * Test method for {@link ShowMapper#mapBack(ShowFO)} with null show.
     */
    @Test
    void mapBack_NullShow() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/mapper/ShowMapperTest.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Show
import cz.vhromada.catalog.web.CatalogMapperTestConfiguration
import cz.vhromada.catalog.web.common.ShowUtils
import cz.vhromada.catalog.web.fo.ShowFO
import cz.vhromada.catalog.web.mapper.impl.ShowMapperImpl
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [Show] and [ShowFO].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogMapperTestConfiguration::class])
class ShowMapperTest {

    /**
     * Mapper for shows
     */
    @Autowired
    private lateinit var mapper: ShowMapper

    /**
     * Test method for [ShowMapperImpl.map].
     */
    @Test
    fun map() {
        val show = ShowUtils.getShow()

        val showFO = mapper.map(show)

        ShowUtils.assertShowDeepEquals(show, showFO)
    }

    /**
     * Test method for [ShowMapperImpl.mapBack].
     */
    @Test
    fun mapBack() {
        val showFO = ShowUtils.getShowFO()

        val show = mapper.mapBack(showFO)

        ShowUtils.assertShowDeepEquals(showFO, show)
    }

}
